src/smart-components/workspaces/managed-selector/TreeViewWorkspaceItem.ts

```diff
@@ -0,0 +1,23 @@
+import { TreeViewDataItem } from '@patternfly/react-core/dist/dynamic/components/TreeView';
+import Workspace from './Workspace';
+
+/**
+ * Extends the TreeViewDataItem to add the underlying workspace that the data
+ * item represents.
+ */
+interface TreeViewWorkspaceItem extends TreeViewDataItem {
+  parentTreeViewItem?: TreeViewWorkspaceItem;
+  workspace: Workspace;
+}
+
+/**
+ * Determines whether the given TreeViewDataItem is in reality a
+ * TreeViewWorkspaceItem or not.
+ * @param treeViewDataItem the item to check.
+ * @returns whether the given element is a TreeViewWorkspaceItem.
+ */
+function instanceOfTreeViewWorkspaceItem(treeViewDataItem: TreeViewDataItem): treeViewDataItem is TreeViewWorkspaceItem {
+  return 'workspace' in treeViewDataItem;
```

Comment:
The `in` checks existence of an attribute but not its value. Is this a valid object?
```js
{
  // rest of the TreeViewDataItem values
  workspaces: null
}
```

Reply:
I think the reason is checking if `TreeViewDataItem`  has the attribute `workspace` is because the TreeView API will naturally use `TreeViewDataItem` but we want to narrow it down to being a `TreeViewWorkspaceItem` when dealing with it.

This code was copied from Mikel's PR.
I don't see it being used anywhere in this PR, probably is used in other parts of said PR.

Reply:
It is being used here: https://github.com/RedHatInsights/insights-chrome/pull/3033/commits/ef15292f8a681ac5e5918d2bf81c6218040fba3f#diff-011eecdeb0bd67e714899b1834c3cd6b31b57e942e9d8fc252791e9f6adfa875R75-R77